packages/react-reconciler/src/__tests__/ReactMemo-test.js

```diff
@@ -497,5 +497,21 @@ describe('memo', () => {
         expect(root).toMatchRenderedOutput('1');
       });
     });
+
+    it('inner component of react memo should have _debugOwner set', async () => {
+      const InComponent = React.forwardRef((props, ref) => <div ref={ref} />);
+      const Outer = React.memo(InComponent);
+      const App = () => {
+        const ref = React.createRef();
+        return <Outer ref={ref}>Click me! </Outer>;
+      };
+      ReactNoop.render(<App />);
+      expect(Scheduler).toFlushWithoutYielding();
+      const innerFiber = ReactNoop.getRoot().current.child.child.child;
```

Comment:
We'd like to avoid tests like this. Is there any behavior that's observable without internal structures?

Maybe there is something testing "Rendered By" in DevTools folder itself?

Reply:
sure yeah it does feel wrong writing test like this. I'll have a look to see if there's somewhere else I can't check.

Reply:
@hanq08 check out the [ownerListContext-test.js](https://github.com/facebook/react/blob/master/packages/react-devtools-shared/src/__tests__/ownersListContext-test.js), I believe that's where you'll want to test this.